### PR TITLE
Layer API Refactoring

### PR DESCRIPTION
**Pull request by DEKHTIARJonathan**
I have reworked and restructured the Layer API.

What do you think about this new organisation ?

### CODE REVIEW

File: tensorlayer/layers/recurrent.py [hunk 1]
```diff
@@ -558,7 +560,7 @@ def __init__(
     @property
     def state_size(self):
         """State size of the LSTMStateTuple."""
-        return (LSTMStateTuple(self._num_units, self._num_units) if self._state_is_tuple else 2 * self._num_units)
+        return LSTMStateTuple(self._num_units, self._num_units) if self._state_is_tuple else (2 * self._num_units)
```
Comment (zsdonghao) [inline tensorlayer/layers/recurrent.py]: are you sure it is the same?

File: tensorlayer/layers/convolution/expert_conv.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+from tensorlayer.decorators import deprecated_alias
+
+__all__ = [
+    'Conv1d',
+    'Conv2d',
+]
+
+
+class Conv1d(Layer):
```
Comment (zsdonghao) [inline tensorlayer/layers/convolution/expert_conv.py]: this is simplified conv, it seems you switched pro APIs and simplified APIs

File: tensorlayer/layers/recurrent.py (hunk 1 above)
Comment (DEKHTIARJonathan) [reply tensorlayer/layers/recurrent.py]: to me it really looks like redundant parentesis... I will revert this change just in doubt ;)
Not a big deal

File: tensorlayer/layers/convolution/expert_conv.py (hunk 2 above)
Comment (DEKHTIARJonathan) [reply tensorlayer/layers/convolution/expert_conv.py]: Oh damned ...
I'll rename the files xD

**Pull request merged**